src/promptflow/promptflow/_core/tracer.py

```diff
@@ -193,35 +254,44 @@ def create_trace(func, args, kwargs):
 
         @functools.wraps(func)
         async def wrapped(*args, **kwargs):
-            if Tracer.active_instance() is None:
-                return await func(*args, **kwargs)  # Do nothing if no tracing is enabled.
-            # Should not extract these codes to a separate function here.
-            # We directly call func instead of calling Tracer.invoke,
-            # because we want to avoid long stack trace when hitting an exception.
-            try:
-                Tracer.push(create_trace(func, args, kwargs))
-                output = await func(*args, **kwargs)
-                return Tracer.pop(output)
-            except Exception as e:
-                Tracer.pop(None, e)
-                raise
+            trace = create_trace(func, args, kwargs)
+            span_name = get_node_name_from_context() if trace_type == TraceType.TOOL else trace.name
```

Comment:
We may have another meeting to review the parameters.

Reply:
Sure.